spec/integration/rails_spec.rb

```diff
@@ -118,211 +104,231 @@ def ping(recipient, subject)
+        expect(service['framework']['name']).to eq 'Ruby on Rails'
+        expect(service['framework']['version'])
+            .to match(/\d+\.\d+\.\d+(\.\d+)?/)
+      end
     end
 
-    it 'prepends Rails.root to log_path' do
-      final_log_path = ElasticAPM.agent.config.log_path.to_s
-      expect(final_log_path).to eq "#{Rails.root}/spec/elastic_apm.log"
+    context 'log path', :allow_running_agent do
+      it 'prepends Rails.root to log_path' do
+        final_log_path = ElasticAPM.agent.config.log_path.to_s
+        expect(final_log_path).to eq "#{Rails.root}/spec/elastic_apm.log"
+      end
     end
 
     describe 'transactions' do
-      it 'spans action and posts it' do
-        get '/'
+      context 'when a simple request is made', :allow_running_agent do
+        it 'spans action and posts it' do
+          get '/'
 
-        wait_for transactions: 1, spans: 2
+          EventCollector.wait_for transactions: 1, spans: 2
 
-        name = @mock_intake.transactions.fetch(0)['name']
-        expect(name).to eq 'ApplicationController#index'
+          name = EventCollector.transactions.fetch(0)['name']
+          expect(name).to eq 'ApplicationController#index'
+        end
       end
 
-      it 'can set tags and custom context' do
-        get '/tags_and_context'
+      context 'when tags and context are set', :allow_running_agent do
+        it 'sets the values' do
+          get '/tags_and_context'
 
-        wait_for transactions: 1, spans: 2
+          EventCollector.wait_for transactions: 1, spans: 2
 
-        context = @mock_intake.transactions.fetch(0)['context']
-        expect(context['tags']).to eq('things' => 1)
-        expect(context['custom']).to eq('nested' => { 'banana' => 'explosion' })
+          context = EventCollector.transactions.fetch(0)['context']
+          expect(context['tags']).to eq('things' => 1)
+          expect(context['custom']).to eq('nested' => {'banana' => 'explosion'})
+        end
       end
 
-      it 'includes user information' do
-        get '/'
+      context 'when there is user information', :allow_running_agent do
+        it 'includes the info in transactions' do
+          get '/'
 
-        wait_for transactions: 1, spans: 2
+          EventCollector.wait_for transactions: 1, spans: 2
 
-        context = @mock_intake.transactions.fetch(0)['context']
-        user = context['user']
-        expect(user['id']).to eq '1'
-        expect(user['email']).to eq 'person@example.com'
+          context = EventCollector.transactions.fetch(0)['context']
+          user = context['user']
+          expect(user['id']).to eq '1'
+          expect(user['email']).to eq 'person@example.com'
+        end
       end
 
-      it 'ignores url patterns' do
-        get '/ping'
-        get '/'
+      context 'when there are ignored url patterns defined', :allow_running_agent do
+        it 'does not create events for the patterns' do
+          get '/ping'
```

Comment:
Did you remove `/ping` from the routes on purpose?

Reply:
No, I didn't. I'll add it back!